Sources/SignalProtocol+Filtering.swift

```diff
@@ -255,15 +275,16 @@ extension SignalProtocol {
     public func take(first count: Int) -> Signal<Element, Error> {
         guard count > 0 else { return .completed() }
         return Signal { observer in
-            var taken = 0
+            let lock = NSRecursiveLock(name: "com.reactive_kit.signal.take")
+            var _taken = 0
             return self.observe { event in
                 switch event {
                 case .next(let element):
-                    if taken < count {
-                        taken += 1
+                    lock.lock(); defer { lock.unlock() }
+                    if _taken < count {
+                        _taken += 1
                         observer.receive(element)
-                    }
-                    if taken == count {
+                    } else if _taken == count {
```

Comment:
@srdanrasic @trupin 

I think this change breaks existing behavior.

This code is work fine with ReactiveKit 3.13.0.
But does not work with ReactiveKit 3.13.1.

```swift
    func testReactiveKitTakeFirst() {
        let expect = expectation(description: #function)
        let observable = Observable(false)
        
        _ = observable.take(first: 1).collect().observeNext { values in
            XCTAssertEqual([false], values)
            expect.fulfill()
        }
        
        waitForExpectations(timeout: 1, handler: nil)
    }
```

Reply:
Thank you @konomae for finding and reporting this. I'm releasing a fix version right now.

Reply:
@srdanrasic Thanks!